fairlearn/preprocessing/_information_filter.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+
+class InformationFilter(BaseEstimator, TransformerMixin):
+    r"""
+    A component that filters out sensitive correlations in a dataset.
```

Comment:
Suggestions for the docstring:
* we have now moved to `numpydoc`
* In the extended summary, we should just describe what the transformation does, and only provide the implementation details in the `Notes` section (see the recommendation [here](https://numpydoc.readthedocs.io/en/latest/format.html#sections))--but in this case we might not even need the implementation notes.
* For the extended description, I was thinking about something like:
```
CorrelationRemover applies a linear transformation to the non-sensitive feature columns in order 
to remove their correlation with the sensitive feature columns while retaining as much information
as possible (as measured by the least-squares error).
```
* If you think it would be useful to add implementation notes, I think that those should contain:
  * statement of the optimization problem that the transformation is solving (minimize the least-squares error subject to the no-correlation constraint)
  * statement that the solution is found by centering sensitive features, fitting the linear regression model to non-sensitive features, and reporting the residual

Reply:
I've added these changes to the docstring now. I'd say it renders nicely. 

![image](https://user-images.githubusercontent.com/1019791/85952190-e7cd8f80-b967-11ea-9c16-851e813be7a5.png)

Reply:
I agree, this looks nice.

Elsewhere we either just say sensitive features or use `A`, not `S`. `Z` as `X \ A` is also new to me. Perhaps we need to standardize on these kinds of things at some point to avoid confusing users? @MiroDudik ?

nit: small typo "mathmatically"